src/InfluxDB.Collector/Util/ITimestampSource.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace InfluxDB.Collector.Util
+{
+    /// <summary>
+    /// Supplier of timestamps for metrics 
+    /// </summary>
+    internal interface ITimestampSource
+    {
+        DateTime GetUtcNow();
```

Comment:
Could this just be a `UtcNow` property?

Reply:
Personally I take the stance that anything that modifies state shouldn't be a property getter. I realize that from a caller point of view, they shouldn't have insight into the implementation details, but this is one of those times where it feels better as a method rather than a getter.

Reply:
I usually follow the same pattern - I did think twice about this, too. On the one hand, signalling it's not just a regular "getter" is useful, but on the other hand, it's a somewhat quirky implementation detail that users should be able to ignore. Happy either way :-)